client/axios-instance.js

```diff
@@ -0,0 +1,114 @@
+const axios = require('axios')
+const jwtDecode = require('jwt-decode')
+const utils = require('./utils')
+
+// env consts
+const BASE_SERVER_URL = process.env.BASE_SERVER_URL || 'http://127.0.0.1:8888/'
+const SERVER_USER_EMAIL = process.env.SERVER_USER_PASSWORD
+const SERVER_USER_PASSWORD = process.env.SERVER_USER_PASSWORD
+
+let accessToken
+let refreshToken
+
+// init axios instance with base url
+const instance = axios.create({
+  baseURL: BASE_SERVER_URL,
+  timeout: 1000
+})
+
+const isTokenAlive = (token) => {
+  if (!token) return false
+
+  let decoded = jwtDecode(token)
+  return decoded.exp > (new Date().getTime() / 1000)
+}
+
+const setTokens = (tokens) => {
+  accessToken = tokens.access
+  refreshToken = tokens.refresh
+}
+
+const getAccessToken = () => {
+  return new Promise((resolve, reject) => {
+    if (!SERVER_USER_EMAIL || !SERVER_USER_PASSWORD) reject(utils.MSG_AUTHORIZATION_DATA_NEEDED)
+
+    return instance.post('/token', { user: { email: SERVER_USER_EMAIL, password: SERVER_USER_PASSWORD } })
+      .then(function (response) {
+        setTokens(response.data)
+        return resolve(response.data)
+      })
+      .catch(() => reject(utils.MSG_AUTHORIZATION_FAILED))
+  })
+}
+
+const refreshAccessToken = (refreshToken) => {
+  return instance.post(`/token/refresh?refresh=${refreshToken}`)
+    .then(function (response) {
+      setTokens(response.data)
+      return Promise.resolve(response.data)
+    })
+    .catch((error) => Promise.reject(error))
+}
+
+const accessTokenCallback = (originalRequest) => {
+  return getAccessToken().then((response) => {
+    originalRequest.params = { access: response.access }
+    return Promise.resolve(originalRequest)
+  })
+    .catch(error => Promise.reject(error))
+}
+
+// Handling the authorization process on requests.
+// refreshing/getting access token if needed.
+
+// request
+instance.interceptors.request.use(config => {
+  const originalRequest = config
+
+  const accessTokenAlive = isTokenAlive(accessToken)
+
+  if ((accessToken && accessTokenAlive) || config.url.match(/^\/token/)) {
+    originalRequest.params = { access: accessToken }
+    return Promise.resolve(originalRequest)
+  } else if (accessToken && !accessTokenAlive) {
+    return refreshAccessToken(refreshToken).then((response) => {
+      originalRequest.params = { access: response.access }
+      return Promise.resolve(originalRequest)
+    })
+      .catch(() => accessTokenCallback(originalRequest))
+  } else {
+    return accessTokenCallback(originalRequest)
+  }
+})
+
+// response
+instance.interceptors.response.use((response) => {
+  return response
+}, (error) => {
+  const originalRequest = error.config
+
+  if (originalRequest._retry) return Promise.reject(error)
```

Comment:
You can use the `rave` function from utils to exclude the error message from the info log stream.